pkg/model/gcemodel/firewall.go

```diff
@@ -116,15 +121,37 @@ func (b *FirewallModelBuilder) Build(c *fi.ModelBuilderContext) error {
 	if b.Cluster.Spec.NonMasqueradeCIDR != "" {
 		// The traffic is not recognized if it's on the overlay network?
 		klog.Warningf("Adding overlay network for X -> master rule - HACK")
+
+		nonMasqueradeCIDR, err := cidr.NewSet([]string{b.Cluster.Spec.NonMasqueradeCIDR})
+		if err != nil {
+			return err
+		}
+
 		t := &gcetasks.FirewallRule{
 			Name:         s(b.SafeObjectName("cidr-to-master")),
 			Lifecycle:    b.Lifecycle,
 			Network:      b.LinkToNetwork(),
-			SourceRanges: []string{b.Cluster.Spec.NonMasqueradeCIDR},
+			SourceRanges: ipv4SourceRange(nonMasqueradeCIDR),
 			TargetTags:   []string{b.GCETagForRole(kops.InstanceGroupRoleMaster)},
 			Allowed:      []string{"tcp:443", "tcp:4194"},
 		}
 		c.AddTask(t)
 	}
 	return nil
 }
+
+func ipv4SourceRange(cidrs cidr.Set) []string {
+	ipv4s := cidrs.WhereIPV4().ToStrings()
+	if len(ipv4s) == 0 {
+		ipv4s = append(ipv4s, "0.0.0.0/32")
+	}
+	return ipv4s
+}
+
+func ipv6SourceRange(cidrs cidr.Set) []string {
+	ipv6s := cidrs.WhereIPV6().ToStrings()
+	if len(ipv6s) == 0 {
+		ipv6s = append(ipv6s, "::/128")
```

Comment:
The semantics here seem odd. If you only list IPv4 CIDRs on your SSH access list, it will open SSH to the entire IPv6 Internet?

Reply:
Actually I think I found one!  We can set disabled=true on the rule - I think that's clearer than both the odd IPs and deleting the rules.  (I _think_!)